### PR TITLE
Spark test timeout config

### PR DESCRIPTION
**Pull request by Petro-Kotsiuba**

### CODE REVIEW

File: integration-tests/src/test/java/com/epam/dlab/automation/test/TestCallable.java [hunk 1]
```diff
@@ -93,8 +97,8 @@ public TestCallable(String notebookTemplate,String dataEngineType, boolean fullT
         LOGGER.info("   SSN provisioned user resources URL is {}", ssnProUserResURL);
     }
 
-    public TestCallable(NotebookConfig notebookConfig, boolean fullTest) {
-    	this(notebookConfig.getNotebook_template(), notebookConfig.getData_engine_type(), fullTest);
+    private static Duration getDuration(String duaration) {
```
Comment (YuriyHolinkoEp) [inline integration-tests/src/test/java/com/epam/dlab/automation/test/TestCallable.java]: spelling

File: integration-tests/examples/config.properties [hunk 2]
```diff
@@ -38,20 +38,21 @@ AWS_REQUEST_TIMEOUT=10s
...
 TIMEOUT_UPLOAD_KEY=40m
-TIMEOUT_NOTEBOOK_CREATE=60m
-TIMEOUT_NOTEBOOK_STARTUP=20m
-TIMEOUT_NOTEBOOK_SHUTDOWN=10m
-TIMEOUT_NOTEBOOK_TERMINATE=10m
-TIMEOUT_EMR_CREATE=60m
-TIMEOUT_EMR_TERMINATE=20m
+TIMEOUT_SSN_STARTUP=60m
+#following timeouts moved to NOTEBOOKS_TO_TEST property
+#TIMEOUT_NOTEBOOK_STARTUP=20m
```
Comment (YuriyHolinkoEp) [inline integration-tests/examples/config.properties]: are these properties used somewhere in the project?

Comment (Petro-Kotsiuba) [reply integration-tests/examples/config.properties]: Commented timeout properties are not used in the project anymore.
Instead NotebookConfig is used. It instantiates from NOTEBOOK_TO_TEST property

File: integration-tests/examples/config.properties [hunk 3]
```diff
@@ -38,20 +38,21 @@ AWS_REQUEST_TIMEOUT=10s
...
+#TIMEOUT_EMR_TERMINATE=20m
 #TIMEOUT_LIB_GROUPS=5m
 #TIMEOUT_LIB_LIST=5m
 #TIMEOUT_LIB_INSTALL=15m
 
 CLUSTER_OS_USERNAME=dlab-user
 CLUSTER_OS_FAMILY=debian
 
-#NOTEBOOKS_TO_TEST=[{"notebook_template":"jupyter","data_engine_type": "dataengine"},{"notebook_template":"rstudio","data_engine_type": "dataengine-service"},{"notebook_template":"zeppelin","data_engine_type": "dataengine"}]
+#NOTEBOOKS_TO_TEST=[{"notebook_template":"jupyter", "data_engine_type":"dataengine", "timeout_notebook_create":"0s", "timeout_notebook_startup":"0s", "timeout_notebook_shutdown":"0s", "timeout_cluster_create":"0s", "timeout_cluster_terminate":"0s", "timeout_lib_groups":"5m", "timeout_lib_list":"5m","timeout_lib_install":"15m"},{"notebook_template":"rstudio","data_engine_type": "dataengine-service"}]
```
[the remark this answers is not in the document]
Comment (Petro-Kotsiuba) [reply integration-tests/examples/config.properties]: Formatted

File: integration-tests/src/test/java/com/epam/dlab/automation/test/TestCallable.java (hunk 1 above)
Comment (Petro-Kotsiuba) [reply integration-tests/src/test/java/com/epam/dlab/automation/test/TestCallable.java]: corrected

File: integration-tests/examples/config.properties [hunk 4]
```diff
@@ -38,20 +38,31 @@ AWS_REQUEST_TIMEOUT=10s
...
 CLUSTER_OS_USERNAME=dlab-user
 CLUSTER_OS_FAMILY=debian
 
-#NOTEBOOKS_TO_TEST=[{"notebook_template":"jupyter","data_engine_type": "dataengine"},{"notebook_template":"rstudio","data_engine_type": "dataengine-service"},{"notebook_template":"zeppelin","data_engine_type": "dataengine"}]
+#NOTEBOOKS_TO_TEST=
+#		[
+#			{
+#				"notebook_template":"jupyter", 
+#				"data_engine_type":"dataengine", 
+#				"timeout_notebook_create":"0s",
```
Comment (YuriyHolinkoEp) [inline integration-tests/examples/config.properties]: Why 0s but not previous default value?

File: integration-tests/examples/config.properties [hunk 5]
```diff
@@ -38,20 +38,31 @@ AWS_REQUEST_TIMEOUT=10s
...
-#TIMEOUT_LIB_GROUPS=5m
-#TIMEOUT_LIB_LIST=5m
-#TIMEOUT_LIB_INSTALL=15m
+TIMEOUT_SSN_STARTUP=60m
 
 CLUSTER_OS_USERNAME=dlab-user
 CLUSTER_OS_FAMILY=debian
 
-#NOTEBOOKS_TO_TEST=[{"notebook_template":"jupyter","data_engine_type": "dataengine"},{"notebook_template":"rstudio","data_engine_type": "dataengine-service"},{"notebook_template":"zeppelin","data_engine_type": "dataengine"}]
+#NOTEBOOKS_TO_TEST=
```
Comment (YuriyHolinkoEp) [inline integration-tests/examples/config.properties]: Can we read notebook configs from formatted view? If no please implement it

File: integration-tests/examples/config.properties [hunk 6]
```diff
@@ -38,20 +38,31 @@ AWS_REQUEST_TIMEOUT=10s
...
+#				"notebook_template":"jupyter", 
+#				"data_engine_type":"dataengine", 
+#				"timeout_notebook_create":"0s", 
+#				"timeout_notebook_startup":"0s", 
+#				"timeout_notebook_shutdown":"0s", 
+#				"timeout_cluster_create":"0s", 
+#				"timeout_cluster_terminate":"0s", 
+#				"timeout_lib_groups":"5m", 
+#				"timeout_lib_list":"5m",
+#				"timeout_lib_install":"15m"
```
Comment (YuriyHolinkoEp) [inline integration-tests/examples/config.properties]: Where is full test property

File: integration-tests/examples/config.properties [hunk 7]
```diff
@@ -38,20 +38,32 @@ AWS_REQUEST_TIMEOUT=10s
...
-#TIMEOUT_LIB_GROUPS=5m
-#TIMEOUT_LIB_LIST=5m
-#TIMEOUT_LIB_INSTALL=15m
+TIMEOUT_SSN_STARTUP=60m
 
 CLUSTER_OS_USERNAME=dlab-user
 CLUSTER_OS_FAMILY=debian
 
-#NOTEBOOKS_TO_TEST=[{"notebook_template":"jupyter","data_engine_type": "dataengine"},{"notebook_template":"rstudio","data_engine_type": "dataengine-service"},{"notebook_template":"zeppelin","data_engine_type": "dataengine"}]
+#NOTEBOOKS_TO_TEST=\
```
[the remark this answers is not in the document]
Comment (Petro-Kotsiuba) [reply integration-tests/examples/config.properties]: Left commented property as an example, but changed value of this property

File: integration-tests/examples/config.properties [hunk 8]
```diff
@@ -38,20 +38,32 @@ AWS_REQUEST_TIMEOUT=10s
...
+#		[\
+#			{\
+#				"notebook_template": "jupyter",\ 
+#				"data_engine_type": "dataengine",\
+#				"full_test": true,\
```
[the remark this answers is not in the document]
Comment (Petro-Kotsiuba) [reply integration-tests/examples/config.properties]: Changed

File: integration-tests/src/main/java/com/epam/dlab/automation/model/NotebookConfig.java [hunk 9]
```diff
@@ -18,34 +18,108 @@
...
+	
+	@JsonProperty("full_test")
+	private boolean fullTest;
+	
+	
+	@JsonProperty("timeout_notebook_create")
+	private String timeoutNotebookCreate="60m";
+	
+	@JsonProperty("timeout_notebook_startup")
+    private String timeoutNotebookStartup="20m";
```
[the remark this answers is not in the document]
Comment (Petro-Kotsiuba) [reply integration-tests/src/main/java/com/epam/dlab/automation/model/NotebookConfig.java]: done

File: integration-tests/src/main/java/com/epam/dlab/automation/model/NotebookConfig.java [hunk 10]
```diff
@@ -18,34 +18,108 @@
...
+	
+	@JsonProperty("timeout_notebook_shutdown")
+    private String timeoutNotebookShutdown="10m";
```
[the remark this answers is not in the document]
Comment (Petro-Kotsiuba) [reply integration-tests/src/main/java/com/epam/dlab/automation/model/NotebookConfig.java]: done

**Pull request merged**